source/blog/2016-07-20-vagrant-service-manager-install-cli.html.md

```diff
@@ -0,0 +1,85 @@
+date: 2016-07-20 09:56 UTC
+tags: atomic-developer-bundle, vagrant-service-manager
+published: true
+comments: true
+---
+
+The [vagrant-service-manager](https://github.com/projectatomic/vagrant-service-manager) plugin is designed to enable easier access to the features and services provided by the [Atomic Developer Bundle (ADB)](https://github.com/projectatomic/adb-atomic-developer-bundle). More information can be found in the README  of 'vagrant-service-manager' repo.
+
+This blog is about the new feature in 'vagrant-service-manger' i.e. `install-cli` command. `install-cli` got released with 'vagrant-service-manager' version 1.2.0. This command installs the client binary for any of the services provided by ADB. At this point it can download client binaries for docker and OpenShift.
+
+Usage:
+
+```
+vagrant service-manager install-cli [service]
+
+Where 'service' can be 'docker' and 'openshift'.
+```
+
+Make sure you have 'vagrant-service-manager' plugin installed in your machine. If not, install it via following command as:
+
+```
+$ vagrant plugin install vagrant-service-manager
+```
+
+Now, ensure it has been installed properly with version 1.2.0 by running following command as:
+
+```
+$ vagrant plugin list
+vagrant-service-manager (1.2.0)
+```
+
+Before proceeding let's setup OpenShift Origin using ADB. To setup, run the following commands:
+
+```
+$ mkdir adb-openshift
+$ cd adb-openshift
+$ curl -o Vagrantfile https://raw.githubusercontent.com/projectatomic/adb-atomic-developer-bundle/master/components/centos/centos-openshift-setup/Vagrantfile
+$ vagrant up
+```
+
+`vagrant up` will take few minutes to 30 minutes as it downloads OpenShift Origin docker images from
+ docker hub. Post the command completion you should see information to access OpenShift setup by ADB.
+
+Now, the OpenShift origin server is ready and you may need a client to access it and perform your
+desired operations. You can manually download the client binary from [OpenShift repository](https://github.com/openshift/origin/releases) but we recommend you to use `install-cli` feature provided by 'vagrant-service-manager'.
+
+You can get start with `install-cli` command as:
+
+```
+$ vagrant service-manager install-cli --help
+```
+
+Now, if you want to install the OpenShift client binary (oc) then run the following command as:
+
+```
+$ vagrant service-manager install-cli openshift
+# Binary now available at /home/budhram/.vagrant.d/data/service-manager/bin/openshift/1.1.1/oc
+# run binary as:
+# oc <command>
+export PATH=/home/budhram/.vagrant.d/data/service-manager/bin/openshift/1.1.1:$PATH
+
+# run following command to configure your shell:
+# eval "$(VAGRANT_NO_COLOR=1 vagrant service-manager install-cli openshift | tr -d '\r')"
+```
+
+Now, the binary has been downloaded and made available at path _/home/budhram/.vagrant.d/data/service-manager/bin/openshift/1.1.1/oc_ and you can configure your shell via command mentioned in the `install-cli` output as:
```

Comment:
Now, the binary has been downloaded and made available in the listed directory.  You can configure your shell with the command mentioned in the output.  This will make sure that the 'oc' binary is in your executable path.